### PR TITLE
devel/sig-scheduling: add scheduler framework plugins doc

### PR DESCRIPTION
**Pull request by adtac**
**Which issue(s) this PR fixes**:

Related to #5237

Moved from https://github.com/kubernetes/kubernetes/blob/35bc0fbad5cd87f1e67fe1f654e616891afbe997/pkg/scheduler/framework/plugins/README.md

cc @Huang-Wei @mrbobbytables

### CODE REVIEW

File: contributors/devel/sig-scheduling/scheduler_framework_plugins.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+# Scheduler Framework Plugins
+
+## Creating a new in-tree plugin
+
+Read [the docs](https://kubernetes.io/docs/concepts/scheduling-eviction/scheduling-framework/)
+to understand the different extension points within the scheduling framework.
+
+TODO([#95156](https://github.com/kubernetes/kubernetes/issues/95156)): flesh this out a bit more.
```
Comment (alculquicondor) [inline contributors/devel/sig-scheduling/scheduler_framework_plugins.md:8]: migrate this issue to community repo

Comment (adtac) [reply contributors/devel/sig-scheduling/scheduler_framework_plugins.md:8]: I don't have the perms to migrate issues I think.

Comment (mrbobbytables) [reply contributors/devel/sig-scheduling/scheduler_framework_plugins.md:8]: I can take care of it 👍

Comment (adtac) [reply contributors/devel/sig-scheduling/scheduler_framework_plugins.md]: Done: https://github.com/kubernetes/community/issues/5466

Thanks, @mrbobbytables!

**Comment by alculquicondor** [comment]
ping @Huang-Wei

File: contributors/devel/sig-scheduling/scheduler_framework_plugins.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	BarParam int32
+}
+```
+
+Note that we embed `k8s.io/apimachinery/pkg/apis/meta/v1.TypeMeta` to include
+API metadata for [versioning and persistence](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#api-conventions).
+We add the `+k8s:deepcopy-gen:interfaces` comment to [auto-generate a `DeepCopy` function](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/code-generator)
+for the struct.
+
+Similarly, define `FooPluginArgs` in `k8s.io/kube-scheduler/config/{version}/types_pluginargs.go`,
```
Comment (Huang-Wei) [inline contributors/devel/sig-scheduling/scheduler_framework_plugins.md:39]: Let's add the prefix "staging/src":

```suggestion
Similarly, define `FooPluginArgs` in `staging/src/k8s.io/kube-scheduler/config/{version}/types_pluginargs.go`,
```

**Review by Huang-Wei — COMMENTED** [review]
Thanks @adtac ! Just one nit.

File: contributors/devel/sig-scheduling/scheduler_framework_plugins.md (hunk 2 above)
Comment (adtac) [reply contributors/devel/sig-scheduling/scheduler_framework_plugins.md:39]: Done.

**Pull request merged**